### PR TITLE
fix displaying enabled/disable mesh dataset group

### LINKED ISSUE #39396: Identify Results of Mesh layer depend on Temporal Controller status?
To me something does not seem to be very consequent when you identify meshes in a Mesh layer:

If I load a Mesh layer (holding more then 20 'groups/layers/dimensions, or what you call those'...
And style one of those groups I want to view.
And activate the identify tool to click in one of the faces, I will see (rightfully) ONLY the activated/visible group.

Screenshot-20201015170644-837x507

But if I make the Temporal Controller active (that netcdf HAS a time dimension), and again identify the same face, the Identify Results panel is filled with ALL groups I have in the netcdf?

Screenshot-20201015170735-835x992

That seems not very consequent? I think that the identify tool should either always ONLY show the visible/active group, OR always ALL groups.
Note that changing the 'Mode' ('Current Layer', 'Top Down' etc) do NOT change the behaviour.
Another note: it IS or can be helpfull to see the face information of all groups!  Maybe an idea would be to show them all in the Top Down* modes? OR create an extra option in the little settings tool?

Or am I missing some deeper insight in the combination of Temporal Controller and Mesh layers?

PS I cannot attach the netcdf here in public, but I'm happy to make it available for anybody if needed.

### PR DESCRIPTION
**Pull request by vcloarec**
In the mesh layer properties table, it is possible to disable dataset groups in the source tab. that allows to no displaying unwanted dataset group.
For now the dataset groups are only disabled in the dataset group tree view used to choose the dataset group to display in the canvas. Dataset groups are not disabled in other different parts of QGIS (mesh calculator, identify map tool, dataset group combo box).
This PR fixes this issue.

Other minor fixes :
- bad updating of dataset groups list from dataset group list model.
- static dataset group not displayed in the identify tool when temporal navigation is off.

fix #39396

### CODE REVIEW

File: src/gui/mesh/qgsmeshdatasetgrouptreeview.cpp [hunk 1]
```diff
@@ -494,17 +494,25 @@ void QgsMeshActiveDatasetGroupTreeView::setActiveGroup()
...
+  endResetModel();
 }
 
 int QgsMeshDatasetGroupListModel::rowCount( const QModelIndex &parent ) const
 {
   Q_UNUSED( parent );
   if ( mRootItem )
-    return mRootItem->totalChildCount();
+  {
+    QList<int> list = mRootItem->enabledDatasetGroupIndexes();
```
Comment (PeterPetrik) [inline src/gui/mesh/qgsmeshdatasetgrouptreeview.cpp:513]: remove unused?

File: src/gui/mesh/qgsmeshdatasetgrouptreeview.cpp [hunk 2]
```diff
@@ -517,7 +525,12 @@ QVariant QgsMeshDatasetGroupListModel::data( const QModelIndex &index, int role
   if ( index.row() >= rowCount( QModelIndex() ) )
     return QVariant();
 
-  QgsMeshDatasetGroupTreeItem *item = mRootItem->childFromDatasetGroupIndex( index.row() );
+  QList<int> list = mRootItem->enabledDatasetGroupIndexes();
```
Comment (PeterPetrik) [inline src/gui/mesh/qgsmeshdatasetgrouptreeview.cpp:528]: const

**Review by PeterPetrik — APPROVED** [review]

File: src/gui/mesh/qgsmeshdatasetgrouptreeview.cpp (hunk 1 above)
Comment (vcloarec) [reply src/gui/mesh/qgsmeshdatasetgrouptreeview.cpp:513]: it is really unused

Comment (PeterPetrik) [reply src/gui/mesh/qgsmeshdatasetgrouptreeview.cpp:513]: so you can remove the line?

Comment (vcloarec) [reply src/gui/mesh/qgsmeshdatasetgrouptreeview.cpp]: ... sorry, I was thinking you talked about the Q_UNUSED... so, of course, removed...

**Pull request merged**